pkg/compactor/compactor.go

```diff
@@ -577,6 +580,12 @@ func (c *Compactor) starting(ctx context.Context) error {
 		return errors.Wrap(err, "failed to start the blocks cleaner")
 	}
 
+	if c.compactorCfg.CachingBucketEnabled {
+		c.bucketClient, err = cortex_tsdb.CreateCachingBucket(c.storageCfg.BucketStore.ChunksCache, c.storageCfg.BucketStore.MetadataCache, c.bucketClient, c.logger, extprom.WrapRegistererWith(prometheus.Labels{"component": "compactor"}, c.registerer))
```

Comment:
I remember we cannot cache deletion marker as it can cause bugs for duplicate compaction. We need to exclude that.

Reply:
Should we remove it for store-gateway and querier too? Just remove the deletion-marker cache completely?